test/api/route/messages_test.dart

```diff
@@ -3,16 +3,177 @@ import 'dart:convert';
 import 'package:checks/checks.dart';
 import 'package:http/http.dart' as http;
 import 'package:test/scaffolding.dart';
+import 'package:zulip/api/model/model.dart';
 import 'package:zulip/api/model/narrow.dart';
 import 'package:zulip/api/route/messages.dart';
 import 'package:zulip/model/narrow.dart';
 
 import '../../example_data.dart' as eg;
 import '../../stdlib_checks.dart';
 import '../fake_api.dart';
+import '../model/model_checks.dart';
 import 'route_checks.dart';
 
 void main() {
+  group('getMessageCompat', () {
+    Future<Message?> checkGetMessageCompat(FakeApiConnection connection, bool expectLegacy, {
+      required int messageId,
+      bool? applyMarkdown,
+    }) async {
+      final result = await getMessageCompat(connection,
+        messageId: messageId,
+        applyMarkdown: applyMarkdown,
+      );
+      expectLegacy
+        ? (check(connection.lastRequest).isA<http.Request>()
+            ..method.equals('GET')
+            ..url.path.equals('/api/v1/messages')
+            ..url.queryParameters.deepEquals({
+              'narrow': jsonEncode([ApiNarrowMessageId(messageId)]),
+              'anchor': messageId.toString(),
+              'num_before': '0',
+              'num_after': '0',
+              if (applyMarkdown != null) 'apply_markdown': applyMarkdown.toString(),
+              'client_gravatar': 'true',
+            }))
+        : (check(connection.lastRequest).isA<http.Request>()
+            ..method.equals('GET')
+            ..url.path.equals('/api/v1/messages/$messageId')
+            ..url.queryParameters.deepEquals({
+              'message_id': messageId.toString(),
+              if (applyMarkdown != null) 'apply_markdown': applyMarkdown.toString(),
+            }));
+      return result;
+    }
+
+    test('modern; message found', () {
+      return FakeApiConnection.with_((connection) async {
+        final message = eg.streamMessage();
+        final fakeResult = GetMessageResult(message: message);
+        connection.prepare(json: fakeResult.toJson());
+        final result = await checkGetMessageCompat(connection, false,
```

Comment:
Let's make `expectLegacy` a named parameter; I think that will make these call sites easier to read, without having to refer back to the definition to remember what the boolean means and which direction it runs in.